app/controllers/appointment_summaries_controller.rb

```diff
@@ -0,0 +1,75 @@
+class AppointmentSummariesController < ApplicationController
+  before_action :require_edit_permission!, except: [:index, :show]
+
+  def index
+    @appointment_summaries = AppointmentSummaries.new(params[:appointment_summaries])
+  end
+
+  def show
+    @appointment_summaries = AppointmentSummaries.new(reporting_month: params[:id])
+  end
+
+  def new
+    @appointment_summary = AppointmentSummary.new(params.permit(:delivery_partner))
+  end
+
+  def create
+    @appointment_summary = AppointmentSummary.new(appointment_summary_params)
+
+    if @appointment_summary.save
+      redirect_to filtered_appointment_summaries(@appointment_summary)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @appointment_summary = AppointmentSummary.find(params[:id])
+  end
+
+  def update
+    @appointment_summary = AppointmentSummary.find(params[:id])
+
+    if @appointment_summary.update(appointment_summary_params)
+      redirect_to filtered_appointment_summaries(@appointment_summary)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    appointment_summary = AppointmentSummary.find(params[:id])
+    flash[:notice] =
+      if appointment_summary.manual?
+        appointment_summary.delete
+        "Deleted appointment summary for #{appointment_summary.descripton}"
+      else
+        "Unable to delete automatically generated appointment summary for #{appointment_summary.descripton}"
+      end
+
+    redirect_to filtered_appointment_summaries(appointment_summary)
+  end
+
+  private
+
+  def appointment_summary_params
+    params.require(:appointment_summary).permit(
+      :delivery_partner,
+      :reporting_month,
+      :transactions,
+      :bookings,
+      :completions
+    ).merge(source: 'manual')
+  end
+
+  def require_edit_permission!
+    authorise_user!('can-edit-data')
```

Comment:
Maybe this permission should better represent the role it describes... Perhaps `analyst` could work here? The fact these are referred to as permissions, when most of the time they actually represent roles is a discussion for another time perhaps 😉